File: backend/src/main/java/com/careermail/controller/GmailController.java
package com.careermail.controller;

import com.careermail.dto.GmailStatusResponse;
import com.careermail.dto.GmailSyncResponse;
import com.careermail.model.entity.ConnectedAccount;
import com.careermail.model.entity.User;
import com.careermail.service.AuthService;
import com.careermail.service.GmailService;
import com.careermail.service.GoogleOAuthService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;
import java.util.Optional;

@RestController
@RequestMapping("/api/gmail")
public class GmailController {

    private final GmailService gmailService;
    private final AuthService authService;
    private final GoogleOAuthService googleOAuthService;

    public GmailController(GmailService gmailService, AuthService authService, GoogleOAuthService googleOAuthService) {
        this.gmailService = gmailService;
        this.authService = authService;
        this.googleOAuthService = googleOAuthService;
    }

    @GetMapping("/status")
    public ResponseEntity<GmailStatusResponse> getStatus() {
        User user = authService.getCurrentUser();
        Optional<ConnectedAccount> accountOpt = gmailService.getConnectedAccount(user);

        boolean configured = googleOAuthService.isConfigured();

        if (accountOpt.isPresent()) {
            ConnectedAccount acc = accountOpt.get();
            return ResponseEntity.ok(new GmailStatusResponse(
                    true,
                    acc.getProviderEmail() != null ? acc.getProviderEmail() : user.getEmail(),
                    acc.getProvider(),
                    acc.getLastSyncedAt(),
                    acc.getTotalEmailsScanned(),
                    configured
            ));
        }

        return ResponseEntity.ok(new GmailStatusResponse(
                false,
                null,
                "google",
                null,
                0,
                configured
        ));
    }

    @PostMapping("/sync")
    public ResponseEntity<GmailSyncResponse> syncGmail(
            @RequestParam(name = "maxResults", defaultValue = "30") int maxResults
    ) {
        User user = authService.getCurrentUser();
        return ResponseEntity.ok(gmailService.scanAndProcess(user, maxResults));
    }

    @PostMapping("/disconnect")
    public ResponseEntity<Map<String, String>> disconnect() {
        User user = authService.getCurrentUser();
        gmailService.disconnectAccount(user);
        return ResponseEntity.ok(Map.of("message", "Gmail integration disconnected successfully"));
    }
}

File: backend/Controllers/GmailController.cs
using CareerMail.Api.Models.DTOs;
using CareerMail.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace CareerMail.Api.Controllers;

[Authorize]
[Route("api/gmail")]
public class GmailController : BaseApiController
{
    private readonly IGmailService _gmailService;
    private readonly IAuthService _authService;
    private readonly IGoogleOAuthService _googleOAuthService;

    public GmailController(IGmailService gmailService, IAuthService authService, IGoogleOAuthService googleOAuthService)
    {
        _gmailService = gmailService;
        _authService = authService;
        _googleOAuthService = googleOAuthService;
    }

    [HttpGet("status")]
    public async Task<ActionResult<GmailStatusResponse>> GetStatus()
    {
        var user = await _authService.GetCurrentUserAsync(CurrentUserId);
        var account = await _gmailService.GetConnectedAccountAsync(user);
        var configured = _googleOAuthService.IsConfigured();

        if (account != null)
        {
            var hasSend = !string.IsNullOrWhiteSpace(account.Scope) &&
                          (account.Scope.Contains("gmail.send") || account.Scope.Contains("mail.google.com"));

            return Ok(new GmailStatusResponse(
                true,
                account.ProviderEmail ?? user.Email,
                account.Provider,
                account.LastSyncedAt,
                account.TotalEmailsScanned,
                configured,
                account.Scope,
                hasSend
            ));
        }

        return Ok(new GmailStatusResponse(
            false,
            null,
            "google",
            null,
            0,
            configured,
            null,
            false
        ));
    }

    [HttpPost("sync")]
    public async Task<ActionResult<GmailSyncResponse>> SyncGmail([FromQuery] int maxResults = 30)
    {
        var user = await _authService.GetCurrentUserAsync(CurrentUserId);
        var result = await _gmailService.ScanAndProcessAsync(user, maxResults);
        return Ok(result);
    }

    [HttpPost("reprocess")]
    public async Task<ActionResult<GmailSyncResponse>> ReprocessStoredEmails()
    {
        var user = await _authService.GetCurrentUserAsync(CurrentUserId);
        var result = await _gmailService.ReprocessStoredEmailsAsync(user);
        return Ok(result);
    }

    [HttpPost("disconnect")]
    public async Task<IActionResult> Disconnect()
    {
        var user = await _authService.GetCurrentUserAsync(CurrentUserId);
        await _gmailService.DisconnectAccountAsync(user);
        return Ok(new { message = "Gmail integration disconnected successfully" });
    }
}
